src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -71,6 +75,18 @@ public void doFilter(ServletRequest request, ServletResponse response,
 		M_log.debug("doFilter: Called");
 		HttpServletRequest useRequest = (HttpServletRequest) request;
 		HttpServletResponse useResponse=(HttpServletResponse)response;
+		
+		if ( BASIC_LTI_LAUNCH_REQUEST.equals(request.getParameter(LTI_MESSAGE_TYPE))) {
+			M_log.debug("new launch so invalidate any existing session");
+			if (useRequest.getSession() != null) {
+				M_log.debug("session id to invalidate: " + useRequest.getSession().getId());
+				useRequest.getSession().invalidate();
+			}
+		}
+		
+		HttpSession session= useRequest.getSession(true);
+		M_log.debug("session id: "+session.getId());
+		
 		if(!checkForAuthorization(useRequest)) {
```

Comment:
I understood that Filter is being called for both service desk and LTI launch... checkForAuthorization() is also being called for LTI launch?